datafusion/src/datasource/memory.rs

```diff
@@ -115,7 +115,7 @@ impl MemTable {
         output_partitions: Option<usize>,
     ) -> Result<Self> {
         let schema = t.schema();
-        let exec = t.scan(&None, batch_size, &[], None)?;
+        let exec = t.scan(&None, batch_size, &[], None, num_cpus::get())?;
```

Comment:
This should not use the number of cpus here, but the default nr of partitions.

Reply:
if I read the code correctly, `MemTable::load` only is used by bench. What's more,  `target_partitions` in `MemTable::scan` is unused.

Reply:
I agree with @Dandandan on this one. Even though `MemTable::scan` is not using `target_partitions`, the passed in table provider `t`'s `scan` method might actually use it. Given `load` is defined as a pub method, I am concerned that there are downstream code calling this method. If it's a private method only used in test, then I think it would be fine to hard code the partition count to cpu cores here.

In short, I recommend adding the partition value as an argument to the `load` method and let caller handle the value resolution.

Reply:
got it!